ui/checkboxradio.js

```diff
@@ -0,0 +1,276 @@
+/*!
+ * jQuery UI Checkboxradio @VERSION
+ * http://jqueryui.com
+ *
+ * Copyright 2014 jQuery Foundation and other contributors
+ * Released under the MIT license.
+ * http://jquery.org/license
+ *
+ * http://api.jqueryui.com/checkboxradio/
+ */
+(function( factory ) {
+	if ( typeof define === "function" && define.amd ) {
+
+		// AMD. Register as an anonymous module.
+		define([
+			"jquery",
+			"./core",
+			"./widget"
+		], factory );
+	} else {
+
+		// Browser globals
+		factory( jQuery );
+	}
+}(function( $ ) {
+
+var formResetHandler = function() {
+		var form = $( this );
+		setTimeout(function() {
+			form.find( ".ui-checkboxradio" ).checkboxradio( "refresh" );
+		});
+	},
+	escapeId = new RegExp( /([!"#$%&'()*+,./:;<=>?@[\]^`{|}~])/g );
+
+$.widget( "ui.checkboxradio", {
+	version: "@VERSION",
+	options: {
+		disabled: null,
+		label: null,
+		icon: true,
+		classes: {
+			"ui-checkboxradio-label": "ui-corner-all",
+			"ui-checkboxradio-icon": "ui-corner-all"
+		}
+	},
+
+	_getCreateOptions: function() {
+		var disabled,
+			that = this,
+			options = this._super() || {};
+
+		this._readLabel();
+
+		this.originalLabel = "";
+		this.label.contents().not( this.element ).each( function() {
+			that.originalLabel += ( this.nodeType === 3 ) ? $( this ).text() : this.outerHTML;
```

Comment:
The only reason i had not opted for something like that is if someone did something crazy like putting multiple inputs inside a single label. Which while completely invalid markup does actually oddly enough work in reality. But I don't think there is any reason for us to support nonsense like that lol.